src/Sylius/Component/Grid/Definition/ArrayToDefinitionConverter.php

```diff
@@ -28,59 +28,103 @@ public function convert($code, array $configuration)
         }
 
         foreach ($configuration['fields'] as $name => $fieldConfiguration) {
-            $field = Field::fromNameAndType($name, $fieldConfiguration['type']);
-
-            if (array_key_exists('path', $fieldConfiguration)) {
-                $field->setPath($fieldConfiguration['path']);
-            }
-            if (array_key_exists('label', $fieldConfiguration)) {
-                $field->setLabel($fieldConfiguration['label']);
-            }
-            if (array_key_exists('enabled', $fieldConfiguration)) {
-                $field->setEnabled($fieldConfiguration['enabled']);
-            }
-            if (array_key_exists('sortable', $fieldConfiguration)) {
-                $field->setSortable($fieldConfiguration['sortable']);
-            }
-            if (array_key_exists('options', $fieldConfiguration)) {
-                $field->setOptions($fieldConfiguration['options']);
-            }
-
-            $grid->addField($field);
+            $grid->addField($this->convertField($name, $fieldConfiguration));
         }
 
         foreach ($configuration['filters'] as $name => $filterConfiguration) {
-            $filter = Filter::fromNameAndType($name, $filterConfiguration['type']);
+            $grid->addFilter($this->convertFilter($name, $filterConfiguration));
+        }
+
+        foreach ($configuration['actions'] as $name => $actionGroupConfiguration) {
+            $grid->addActionGroup($this->convertActionGroup($name, $actionGroupConfiguration));
+        }
+
+        return $grid;
+    }
+
+    /**
+     * @param string $name
+     * @param array $configuration
+     *
+     * @return Field
+     */
+    private function convertField($name, array $configuration)
+    {
+        $field = Field::fromNameAndType($name, $configuration['type']);
+
+        if (array_key_exists('path', $configuration)) {
```

Comment:
Actually, I think that these keys should be always set (configuration parser should be responsible), but since it isn't, calling **known** methods with usage of `PropertyAccessor` rather obfuscates it than clears it.